lib/vast/CodeGen/DefaultVisitor.cpp

```diff
@@ -21,16 +23,25 @@ namespace vast::cg
         , clang::CUDAGlobalAttr
     >;
 
-    operation default_visitor::visit_decl_attrs(operation op, const clang_decl *decl, scope_context &scope) {
+    operation default_visitor::visit_decl_attrs(
+        operation op, const clang_decl *decl, scope_context &scope
+    ) {
         if (decl->hasAttrs()) {
             mlir::NamedAttrList attrs = op->getAttrs();
             for (auto attr : exclude_attrs< excluded_attr_list >(decl->getAttrs())) {
                 auto visited = self.visit(attr, scope);
 
-                auto key = visited.getAbstractAttribute().getName();
+                // All attributes in unsupported dialect have the same name
+                auto is_unsup = mlir::isa< unsup::UnsupportedDialect >(visited.getDialect());
```

Comment:
This should be fixed on the side of unsupported dialect . It leaks the unsupported dialect here, even though it might not be used.

Reply:
create issue for this for later please